app_dart/lib/src/service/scheduler.dart

```diff
@@ -460,8 +460,9 @@ class Scheduler {
             pullRequest: pullRequest,
             checkRunGuard: '$lock',
             logCrumb: logCrumb,
+
+            // The if-branch already skips the engine build phase.
```

Comment:
Is the branch you are talking about here this one -> `isFusion && await _applyFrameworkOnlyPrOptimization(`